e2e_tests/tests/all_providers/normal.rs

```diff
@@ -187,27 +191,47 @@ fn invalid_provider_list_clients() {
 fn list_and_delete_clients() {
     let mut client = TestClient::new();
     client.do_not_destroy_keys();
-    client.set_default_auth(Some("list_clients test".to_string()));
+
+    let all_providers_user = "list_clients test".to_string();
+    client.set_default_auth(Some(all_providers_user.clone()));
 
     let clients = client.list_clients().expect("list_clients failed");
-    assert!(!clients.contains(&"list_clients test".to_string()));
+    assert!(!clients.contains(&all_providers_user));
+
+    let providers = client.list_providers().expect("Failed to list providers");
+    let mut suitable_providers = vec![];
 
-    let key1 = String::from("list_clients1");
-    let key2 = String::from("list_keys2");
-    let key3 = String::from("list_keys3");
+    for provider in providers.iter() {
+        client.set_provider(provider.id);
+        if !client.is_operation_supported(Opcode::PsaGenerateKey) {
+            continue;
+        }
+        suitable_providers.push(provider.clone());
 
-    client.set_provider(ProviderID::MbedCrypto);
-    client.generate_rsa_sign_key(key1.clone()).unwrap();
-    client.set_provider(ProviderID::Pkcs11);
-    client.generate_rsa_sign_key(key2.clone()).unwrap();
-    client.set_provider(ProviderID::Tpm);
-    client.generate_rsa_sign_key(key3.clone()).unwrap();
+        client.set_default_auth(Some(all_providers_user.clone()));
+        client
+            .generate_rsa_sign_key("all-providers-user-key".to_string())
+            .unwrap();
+
+        client.set_default_auth(Some(format!("user_{}", provider.id)));
+        client
+            .generate_rsa_sign_key(format!("user_{}-key", provider.id))
+            .unwrap();
+    }
+
+    client.set_default_auth(Some(all_providers_user.clone()));
 
     let clients = client.list_clients().expect("list_clients failed");
-    assert!(clients.contains(&"list_clients test".to_string()));
-    client
-        .delete_client("list_clients test".to_string())
-        .unwrap();
+    assert_eq!(clients.len(), suitable_providers.len() + 1);
```

Comment:
Might there be concurrency problems if that test is executed with other ones creating keys at the same time with other auth values?
If so, maybe we can check that `clients` contains (at least) exactly what we expect.
edit: oh it's actually just what you are doing in the next lines, do we need this `assert` line 225 then?

Reply:
Yeah, I was thinking that we need to check we have the exact set that we're expecting but yeah, it makes sense to only look for the ones we created in this test. Will remove this assert.